### DISCUSSION TITLE
Public docker image

### DISCUSSION THREAD

**Discussion by link-webcreations** [ParisNeo/lollms-webui#192] [General]
Hi All,

It would be great if we could have a publicly available image already build!
I can see that there is already a Github Action to build and lint, could we consider adding a push ?

Have a good day !

**Comment by ParisNeo** [comment]
Hi there, If you want, you can do this yourself. I accept contribution from the community. Just fork the project, add your action, do a pull request and I'll check it and validate it.

**Reply by SheepIsGoat** [reply]
@ParisNeo I'll hop on this, I just did a bunch of refactoring to my local to make Docker containers work pretty smoothly. I'll fork and PR, but I have to make sure nothing is broken first. Just let me know if any of the refactors are too big before I test it all thoroughly.

On a  related note, if you're open for releasing breaking changes, relying on docker as the default launch method would open the way to a linux-only codebase, which would be way easier to maintain going forward than continuing to support all 3 major OS platforms.

**Reply by ParisNeo** [reply]
Hi there.
Thanks for your interest. As I said, I am welcoming any contribution. But I do want to keep the app usable by non hacky people. It is cool to have a docker install method, but i Need to keep the windows and macos installs also. The aim of lollms is to provide an easy to use app for all. I can't afford loosing newbie users. I have seen what happens when I make it complicated with the petals binding which is linux only. I had to make a wsl version for windows and I see that no one is using it.
Lollms-webui still needs a simple and stable install procedure for all three major os. It is not a multi-user app by essence to be installed on a server. I would build a docker version for lollms (the raw lollms, not the webui) and run the gandalf server. Gandalf has the same service style as open ai api to be compatible with all major apps. It can be installed somewhere and get requests from lollms-webui front end or other tools like autogpt and others. It has something openai hasn't which is access to personnalities directly from the API. It is under construction and is not yet committed to pypi but it should be ready this week.

**Reply by SheepIsGoat** [reply]
@ParisNeo Sounds great, hopefully the docker version will be useful to some, I forgot how painful docker can be on windows. I just put in my PR for you to go over; there's a lot lines of changes, but it's all either refactoring or Docker related. Please feel free to ask questions or modifications on any parts. Also, if you like my style and want any extra hands on Gandalf let me know.

**Reply by ParisNeo** [reply]
Thank you.
I'll take a look at it as soon as I have time.

**Reply by ParisNeo** [reply]
I took a fast view on you changes. They seem promissing. I would just ask for a little favor.
If you have time to make a little document where you explain how the app.py was split into this new structure it would be great for future coding.
Also, Yesterday I added the code for extensions which will be a very cool feature in the future. Please make sure they are updated in your fork before i do a merge.
Great job so far. I'll take a deeper look at it this evening.

Thanks so much for your contribution.

**Reply by SheepIsGoat** [reply]
@ParisNeo Got it, the PR is updated with that now. I added some readme stuff on Docker and some of the technical how the app.py got split. The folder names themselves are not particularly meticulously chosen. I just read through and decided where to put the methods, which is essentially just tagging them with via directory names. It has no effect on functionality, and I'd expect stuff to get moved around and reorganized as you/we learn the best way to organize stuff.

**Reply by ParisNeo** [reply]
I have started the merge but unfortunately the github actions fail. Can you verify that the code passes all tests?

**Comment by ParisNeo** [comment]
Cool. Can you do a pull request on experiments branch. I can take care of upgrading your struture before merging it all to the main branch.

**Comment by DrGood01** [comment]
Salut,

J'aimerais bien échanger brièvement avec toi au téléphone.
Comme dit en objet, je suis maçon. Mais je suis aussi geek. Avec plein
d'enfants. Je cherche à développer une ia familiale aimante aimée et
protectrice; la montagne de code que tu as pondu m'est très utile, mais
j'ai pu remarquer au fil du temps des changements drastiques et des
revirements dans les façons d'installer, bref. encore une fois j'aimerais
bien en parler brièvement je ne veux pas te prendre trop de temps.

Mon explant : 0651477938

Et si tu ne m'appelles jamais,

Bonne vie!

Dominique Godon
Maçon du bâti ancien

Mailher
19700 Lagraulière

0651477938
***@***.***



Le dim. 1 oct. 2023 à 06:35, Saifeddine ALOUI ***@***.***> a
écrit :

**Comment by ParisNeo** [comment]
Bonjour, vous n'êtes pas obligé de donner vos coordonnées sur un site publique.
Je vous conseille de supprimer le message et venir sur Discord. On pourra parler si vous avez des questions:
https://discord.gg/kBAacUE8

C'est gratuit et je peux vous répondre en temps réel les soirs à partir de 20h.
Je suis cool et je snobe pas les gens :)
